dds_cli/data_putter.py

```diff
@@ -169,6 +170,85 @@ def put(
                         for x in [y.id for y in progress.tasks if y.fields.get("step") != "put"]
                     ]
 
+        # Make a single database update for files that have failed
+        # Json file for failed files should only be created if there has been an error
+        if putter.failed_delivery_log.is_file():
```

Comment:
We need to encapsulate this entire thing in a `try/except`, will make it easier to cancel the operation without a bunch of `try/except`s in this code, and we should just be able to raise the exceptions instead of doing `try: except: else:` in the places we need to catch them.

Start with this and then I can explain in another review what I mean about the other try blocks